lib/librte_regexdev/rte_regexdev.c

```diff
@@ -2,4 +2,133 @@
  * Copyright(C) 2019 Marvell International Ltd.
  */
 
-#include <rte_regexdev.h>
+#include <string.h>
+
+#include <rte_spinlock.h>
+#include <rte_memory.h>
+#include <rte_memcpy.h>
+#include <rte_memzone.h>
+#include <rte_string_fns.h>
+
+#include "rte_regexdev.h"
+#include "rte_regexdev_driver.h"
+
+/* Shared memory between primary and secondary processes. */
+static struct {
+	uint64_t next_owner_id;
+	rte_spinlock_t ownership_lock;
+	struct rte_regex_dev_data data[RTE_MAX_REGEXDEV_PORTS];
+} *rte_regex_dev_shared_data;
+
+static const char *MZ_RTE_REGEX_DEV_DATA = "rte_regex_dev_data";
+
+struct rte_regex_dev rte_regex_devices[RTE_MAX_REGEXDEV_PORTS];
+
+/* spinlock for shared data allocation */
+static rte_spinlock_t rte_regex_shared_data_lock = RTE_SPINLOCK_INITIALIZER;
+
+
+static uint16_t
+regex_dev_find_free_port(void)
+{
+	unsigned i;
+
+	for (i = 0; i < RTE_MAX_REGEXDEV_PORTS; i++) {
+		if (rte_regex_devices[i].data != NULL &&
+		    rte_regex_devices[i].data->name == '\0')
+			return i;
+	}
+	return RTE_MAX_REGEXDEV_PORTS;
+}
+
+static struct rte_regex_dev*
+regex_dev_allocated(const char *name)
+{
+	unsigned i;
+
+	for (i = 0; i < RTE_MAX_REGEXDEV_PORTS; i++) {
+		if (rte_regex_devices[i].data != NULL &&
+		    strcmp(rte_regex_devices[i].data->name, name) == 0)
+			return &rte_regex_devices[i];
+	}
+	return NULL;
+}
+
+static struct rte_regex_dev*
+regex_dev_get(uint16_t port_id)
+{
+	struct rte_regex_dev *regex_dev = &rte_regex_devices[port_id];
+
+	regex_dev->data = &rte_regex_dev_shared_data->data[port_id];
```

Comment:
Don't really like that get function is not const, and changing internal structs.
looks more like init, or alloc

Reply:
@orikam yeah, function should not write data pointer, or there should be regex_dev_put() function too.

However this function is only called once, just inline the code and get rid of the function.
or function name should be assign_data()

Reply:
This function will be called much more then once.

I think that like @paravmellanox I will set the data inline and the git will just return a pointer to the regex device. what do you think?